src/OpenFeature.Contrib.Providers.Flagd/Resolver/Rpc/RpcResolver.cs

```diff
@@ -0,0 +1,492 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Security;
+using System.Threading.Tasks;
+using System.Security.Cryptography.X509Certificates;
+using Google.Protobuf.WellKnownTypes;
+using Grpc.Core;
+using Grpc.Net.Client;
+using OpenFeature.Error;
+using OpenFeature.Flagd.Grpc;
+using OpenFeature.Model;
+using ProtoValue = Google.Protobuf.WellKnownTypes.Value;
+using System.Net.Sockets;
+using System.Threading;
+using Value = OpenFeature.Model.Value;
+
+namespace OpenFeature.Contrib.Providers.Flagd.Resolver.Rpc
+{
+    internal class RpcResolver : Resolver
+    {
+        static int EventStreamRetryBaseBackoff = 1;
+        private readonly FlagdConfig _config;
+        private readonly ICache<string, object> _cache;
+        private readonly Service.ServiceClient _client;
+        private readonly System.Threading.Mutex _mtx;
+        private int _eventStreamRetries;
+        private int _eventStreamRetryBackoff = EventStreamRetryBaseBackoff;
+
+        internal RpcResolver(FlagdConfig config)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException(nameof(config));
+            }
+
+            _config = config;
+
+            _client = BuildClientForPlatform(_config);
+            _mtx = new Mutex();
+
+            if (_config.CacheEnabled)
+            {
+                _cache = new LRUCache<string, object>(_config.MaxCacheSize);
+            }
+        }
+
+        internal RpcResolver(Service.ServiceClient client, FlagdConfig config, ICache<string, object> cache = null) : this(config)
+        {
+            _client = client;
+            _cache = cache;
+        }
+
+        public void Init()
+        {
+            if (_config.CacheEnabled)
+            {
+                var handleEvents = new Thread(HandleEvents);
+                handleEvents.Start();
+            }
+        }
+
+        public void Shutdown()
+        {
+            throw new System.NotImplementedException();
```

Comment:
I think we should handle this as we now support init and shutdown

Reply:
I've implemented this now, see commit [here](https://github.com/open-feature/dotnet-sdk-contrib/pull/149/commits/a0e86661f7dc3744e12a3a05f0a37f02a3d16b97).

cc @bacherfl